astropy/coordinates/tests/test_representation.py

```diff
@@ -519,25 +569,46 @@ def test_representation_shortcuts(self):
 
     def test_transform(self):
 
-        s1 = UnitSphericalRepresentation(lon=[1, 2] * u.deg, lat=[3, 4] * u.deg)
+        # set up representation
+        ds1 = UnitSphericalDifferential(d_lon=[1, 2] * u.mas / u.yr,
+                                        d_lat=[3, 4] * u.mas / u.yr,)
+        s1 = UnitSphericalRepresentation(lon=[1, 2] * u.deg, lat=[3, 4] * u.deg,
+                                         differentials=ds1)
 
         matrix = rotation_matrix(-10, "z", u.deg)
 
+        # transform representation & get comparison (thru CartesianRep)
         s2 = s1.transform(matrix)
+        ds2 =  s2.differentials["s"]
+
+        dexpected = UnitSphericalDifferential.from_cartesian(
```

Comment:
As above, answers are known for this case.

Reply:
Done! added checks against ds1